examples/todo.rs

```diff
@@ -115,35 +123,40 @@ fn clear(database: &Database, _context: Context, _response: Response) {
 
 //Send one particular to-do, selected by its id
 fn get_todo(database: &Database, context: Context, mut response: Response) {
-    let host = or_abort!(context.headers.get(), response, StatusCode::BadRequest);
+    let host = or_abort!(context.headers.get(), response, MISSING_HOST_HEADER);
 
     let id = or_abort!(
         context.variables.parse("id").ok(),
         response,
-        StatusCode::BadRequest
+        BAD_ID
     );
 
     let todo = database.read().unwrap().get(id).map(|todo| {
         NetworkTodo::from_todo(&todo, host, id)
     });
 
-    response.send(json::encode(&todo).unwrap());
+    match todo {
+      Some(todo) => response.send(json::encode(&todo).unwrap()),
+      None =>  {
+        response.set_status(StatusCode::NotFound);
+      }
+    };
```

Comment:
This may actually be wrong according to the specification. I'll have to test it to know, because it's test driven without a written specification...